tests/handlers/test_federation.py

```diff
@@ -190,6 +199,101 @@ def test_rejected_state_event_state(self):
 
         self.assertEqual(sg, sg2)
 
+    def test_floating_outlier_membership_auth(self):
+        """
+        Check that we can properly process an event with auth_events that
+        include a floating membership event.
+
+        Regression test for #TODO(PR).
+        """
+        OTHER_SERVER = "otherserver"
+        OTHER_USER = "@otheruser:" + OTHER_SERVER
+
+        # create the room
+        user_id = self.register_user("kermit", "test")
+        tok = self.login("kermit", "test")
+        room_id = self.helper.create_room_as(room_creator=user_id, tok=tok)
+        room_version = self.get_success(self.store.get_room_version(room_id))
+
+        prev_event_ids = self.get_success(self.store.get_prev_events_for_room(room_id))
+        (
+            most_recent_prev_event_id,
+            most_recent_prev_event_depth,
+        ) = self.get_success(self.store.get_max_depth_of(prev_event_ids))
+        # mapping from (type, state_key) -> state_event_id
+        prev_state_map = self.get_success(
+            self.state_store.get_state_ids_for_event(most_recent_prev_event_id)
+        )
+        # List of state event ID's
+        prev_state_ids = list(prev_state_map.values())
+        auth_event_ids = prev_state_ids
+
+        auth_events = self.get_success(self.store.get_events(auth_event_ids))
+        logger.info("auth_event_ids %s\nauth_events=%s", auth_event_ids, auth_events)
+
+        # build a floating outlier member state event
+        fake_prev_event_id = "$" + random_string(43)
+        member_event = event_from_pdu_json(
+            {
+                "type": EventTypes.Member,
+                "content": {
+                    "membership": "join",
+                },
+                "state_key": OTHER_USER,
+                "room_id": room_id,
+                "sender": OTHER_USER,
+                "depth": most_recent_prev_event_depth,
+                "prev_events": [fake_prev_event_id],
+                "auth_events": auth_event_ids.copy(),
+                "origin_server_ts": self.clock.time_msec(),
+                "signatures": {
+                    OTHER_SERVER: {"ed25519:key_version": "SomeSignatureHere"}
+                },
+            },
+            room_version,
+        )
```

Comment:
TODO: Need to fix the `403: No create event in auth events` problem when this test runs

Reply:
This was caused by just returning `[member_event]` in the `self.handler.federation_client.get_event_auth` stub